humphrey/src/tokio/app.rs

```diff
@@ -129,51 +133,63 @@ where
         let error_handler = Arc::new(self.error_handler);
 
         loop {
-            match socket.accept().await {
-                Ok((mut stream, _)) => {
-                    let cloned_state = self.state.clone();
-
-                    // Check that the client is allowed to connect
-                    if (self.connection_condition)(&mut stream, cloned_state) {
-                        let cloned_state = self.state.clone();
-                        let cloned_monitor = self.monitor.clone();
-                        let cloned_subapps = subapps.clone();
-                        let cloned_default_subapp = default_subapp.clone();
-                        let cloned_error_handler = error_handler.clone();
-
-                        cloned_monitor.send(
-                            Event::new(EventType::ConnectionSuccess)
-                                .with_peer_result(stream.peer_addr()),
-                        );
-
-                        // Spawn a new thread to handle the connection
-                        tokio::spawn(async move {
-                            cloned_monitor.send(
-                                Event::new(EventType::ThreadPoolProcessStarted)
-                                    .with_peer_result(stream.peer_addr()),
-                            );
+            let shutdown = async {
+                if let Some(sd) = self.shutdown.clone() {
```

Comment:
Can we change the `.clone()` to `.as_ref()`? Just to avoid the unnecessary allocation.